lib/cache/expirations/AbsoluteExpiration.ts

```diff
@@ -0,0 +1,27 @@
+import { Expiration } from './Expiration';
+
+export class AbsoluteExpiration implements Expiration {
+
+  private readonly expirations = new Set<string>();
+
+  constructor(
+    private readonly timeout: number,
+  ) { }
+
+  public add(key: string, clear: () => any): void {
+    if (this.expirations.has(key)) {
+      return;
+    }
+
+    this.expirations.add(key);
+    setTimeout(this.clear(key, clear), this.timeout);
+  }
+
+  private clear(key: string, clear: () => any): () => void {
+    return () => {
+      clear();
```

Comment:
`clear` is an external function that might fail, in which case we're introducing a memory leak. I'd suggest that we first delete the key, and then invoke `clear()`.